[DRAFT] matter: samples: power consumption on debug (#24587)
nrfconnect/sdk-nrf: doc/nrf/protocols/matter/end_product/platform_and_dmp.rst

.. _ug_matter_platform_and_dmp:

Matter platform and Derived Matter Product (DMP)
###############################################

.. contents::
   :local:
   :depth: 2

> Program summary

Reviewer: People like images. Maybe we can create one for this section to show the whole process? We have something similar for certification here: https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/protocols/matter/end_product/certification.html#certification_process_overview

What do you think @peknis?

Reviewer 2: Sure, images are often helpful and give information at a quick glance. If you decide to draw an image, please use Visio. Do not attach the original .vsd file, but the .svg in the PR. Make sure to have the original stored safely and when ready, ask me to save the original to the place where we store them for potential later adjustments.

***************

Matter platform certification (trial program)
=============================================

Matter Platform Certification validates foundational Matter functionality that can be reused across products.
A certified platform implements supported transports (Thread, Wi‑Fi, Ethernet), commissioning, the Root Node device type and associated baseline clusters, and documents its Supported Operating Environment (SOE).
The platform is evaluated once by an Authorized Test Laboratory (ATL) using the CSA Test Harness against platform‑only and both tests.

> Policy notes:

Reviewer: Maybe put it into `.. note::`?

----------------------

* Faster path to certification by reusing platform test results and only running product‑only plus both tests.
* Lower cost and effort due to a smaller test scope.
* Quality and interoperability by building on a validated foundation (commissioning, secure channel, baseline clusters).
* Focus on product value: device type behavior, application clusters, UX.

What a DMP must do
-------------------

* Match the platform’s Matter specification version.
* Stay within the platform’s SOE (boards/radios, SDK/OS versions).
* Provide the Matter Security Attestation (covering both platform and product).
* Declare product PICS that are a subset or compatible with the platform’s certified capabilities in core functionalities.

Supported Operating Environment (SOE)
=====================================

The Matter Platform SOE defines hardware families/boards and software versions (SDK/OS) supported by the platform.
Staying within the Matter Platform SOE and the platform’s certified scope preserves test skipping at DMP time.

Matter platform: application and testing
***************************************

Prerequisites
=============

* CSA membership in good standing.

> * Implement at least one transport (Thread, Wi‑Fi, or Ethernet).

Reviewer: ```suggestion
* Implement at least one transport layer dedicated to the Matter protocol (Thread, Wi‑Fi, or Ethernet).
```

> * Support Matter commissioning (on‑network required).

Reviewer: ```suggestion
* Support Matter on‑network commissioning.
```

* Application code: device type logic, endpoints, attributes, UI/UX, application cluster handlers.
* App configuration: Kconfig/DTS overlays for product peripherals.
* Optional clusters/features within platform scope (enable/disable) with matching PICS updates.
* Manufacturing data and branding: SKUs, product strings, documentation.
* Fixes to application clusters in ``modules/lib/matter`` (not part of platform).

Prepare certification inputs
============================

* Product PICS with ``PLAT.CERT.TESTS.DONE = True``.
* ``dmp-test-skip.xml`` from the PICS Tool (used by Test Harness to skip platform‑only tests).
* Ensure product PICS remains a subset or compatible with platform capabilities in core functionalities.
* Request the platform’s templated security attestation (pre‑filled with platform details) via DevZone.
* Leverage dependent certification identifiers from platform:
    * Thread <LINK TO COMPATIBILITY MATRIX>
    * Bluetooth <LINK TO COMPATIBILITY MATRIX>
* Retrieve the Platform PICS baseline from the CSA website <LINK TO PICS BASELINE> to confirm compatibility.

Pre-validation using the CSA Test Harness
=========================================

> It is recommended (but optional) to perform a pre-validation of your product test cases using the CSA Test Harness before engaging with the Authorized Test Laboratory (ATL).

Reviewer: ```suggestion
It is recommended to perform a pre-validation of your product test cases using the CSA Test Harness before engaging with the Authorized Test Laboratory (ATL).
```

.. _ug_matter_platform_and_dmp:

> Matter platform and Derived Matter Product (DMP)
> ###############################################

Reviewer 2: ```suggestion
Matter platform and Derived Matter Product (DMP)
################################################
```

To retain test inheritance from the platform, avoid the following changes:

* Upgrading ``modules/lib/matter`` beyond the tag corresponding to the certified |NCS| tag without a coordinated platform update.
* Changing radio/PHY parameters, Wi‑Fi bands, or Thread version beyond the SOE.
* Editing the platform PICS baseline or any platform test list artifacts.

What you can change safely (typical product work)
=================================================

You can safely make the following product‑level changes:

* Application code: device type logic, endpoints, attributes, UI/UX, application cluster handlers.
* App configuration: Kconfig/DTS overlays for product peripherals.
* Optional clusters/features within platform scope (enable/disable) with matching PICS updates.
* Manufacturing data and branding: SKUs, product strings, documentation.
* Fixes to application clusters in ``modules/lib/matter`` (not part of platform).

Prepare certification inputs
============================

Before testing, prepare the following inputs:

* Product PICS with ``PLAT.CERT.TESTS.DONE = True``.
* ``dmp-test-skip.xml`` from the PICS Tool (used by Test Harness to skip platform‑only tests).

> * Ensure product PICS remains a subset or compatible with platform capabilities in core functionalities.

Reviewer 2: ```suggestion
* Ensure that the product PICS remains a subset of or compatible with platform capabilities in core functionalities.
```

> * Request the platform’s templated security attestation (pre‑filled with platform details) via DevZone.

Reviewer 2: ```suggestion
* Request the platform’s templated security attestation (pre‑filled with platform details) through the `DevZone`_.
```

* Leverage dependent certification identifiers from platform:
    * Thread <LINK TO COMPATIBILITY MATRIX>
    * Bluetooth <LINK TO COMPATIBILITY MATRIX>
* Retrieve the Platform PICS baseline from the CSA website <LINK TO PICS BASELINE> to confirm compatibility.

Pre-validation using the CSA Test Harness
=========================================

It is recommended (but optional) to perform a pre-validation of your product test cases using the CSA Test Harness before engaging with the Authorized Test Laboratory (ATL).

> In the Test Harness, create a DMP project, upload your PICS, and add the ``dmp-test-skip.xml`` file.

Reviewer 2: ```suggestion
In the Test Harness, create a DMP project, upload your PICS, and add the :file:`dmp-test-skip.xml` file.
```

The Test Harness will automatically select the product-only tests and all "both" tests to be re-run.
This pre-validation helps ensure your PICS and test setup are correct and can identify issues early, streamlining the formal certification process.

For detailed Test Harness setup instructions for DMP projects, see the Matter Test Harness Guide, available in the `CSA Matter Resource Kit`_ in Matter Test Harness section.

> Use the following steps to prepare the artifacts and configure the Test Harness:

Reviewer 2: ```suggestion
Complete the following steps to prepare the artifacts and configure the Test Harness:
```

1. Generate ``dmp-test-skip.xml`` in the PICS Tool (Platform mode).

   Start by preparing the skip list that instructs the Test Harness to omit platform-only tests.

>    a. Open the PICS Tool and switch to Platform mode.
>    b. Import the certified platform PICS baseline (for example, ``base.xml``) to fingerprint the platform capabilities.
>    c. Create or import your product PICS and complete the product metadata (name, configuration).
>    d. Run PICS validation against the platform baseline to confirm compatibility in core functionalities.
>    e. Export the validated product PICS and the generated ``dmp-test-skip.xml`` file and test case list.

Reviewer 2: ```suggestion
   a. Open the PICS Tool and switch to Platform mode.
   #. Import the certified platform PICS baseline (for example, :file:`base.xml`) to fingerprint the platform capabilities.
   #. Create or import your product PICS and complete the product metadata (name, configuration).
   d. Run PICS validation against the platform baseline to confirm compatibility in core functionalities.
   e. Export the validated product PICS and the generated :file:`dmp-test-skip.xml` file and test case list.
```

2. Create a DMP project in the Test Harness and load artifacts.

   Next, configure the Test Harness to use your validated PICS and the skip file.

   a. Create a new DMP project in the Test Harness.
   b. Upload the validated product PICS.
   c. Add the ``dmp-test-skip.xml`` generated by the PICS Tool.
   d. Execute the test case (based on the test case list from the PICS Tool) and review results prior to ATL engagement.

Arrange testing and apply for certification with the ATL
========================================================

After pre-validation, coordinate with your chosen ATL to schedule the official certification testing.

> Provide the ATL with your PICS, ``dmp-test-skip.xml``, and any required documentation.

Reviewer 2: ```suggestion
Provide the ATL with your PICS, :file:`dmp-test-skip.xml` file, and any required documentation.
```

After approval, register your product in the DCL and follow the logo usage guidelines.

DMP submission checklist
************************

Use this checklist to prepare and submit your DMP efficiently:

[ ] Board and transport selection confirmed within SOE and certified |NCS| tag noted.
[ ] Prepare the product partition layout and bootloader configuration (see :ref:`ug_matter_device_bootloader_partition_layout` and :ref:`ug_matter_device_bootloader`).
[ ] Generate test or get real Attestation Certificates (see :ref:`ug_matter_device_attestation`).
[ ] Factory data prepared (VID, PID, discriminator, etc.); onboarding codes generated if needed (see :ref:`_ug_matter_device_factory_provisioning`).
[ ] Generated Certification Declaration for the product for certification testing (see :ref:`ug_matter_device_configuring_cd`).
[ ] Enable Test Event Triggers for the product for certification testing (see :ref:`ug_matter_device_test_event_trigger`).
[ ] Versioning aligned (MCUboot image version and Matter software version string) for OTA over Matter if applicable (see :ref:`ug_matter_device_versioning`).
[ ] Product PICS exported and validated in the PICS Tool.

> [ ] ``dmp-test-skip.xml`` generated from the PICS Tool to skip platform‑only tests.

Reviewer 2: ```suggestion
[ ] The :file:`dmp-test-skip.xml` file generated from the PICS Tool to skip platform‑only tests.
```

[ ] Security attestation prepared (platform + product), consistent with the platform’s certified scope (see :ref:`ug_matter_device_security` for more information on security in Matter).
[ ] Dependent certification evidence collected as applicable based on the platform's certification (see :ref:`ug_matter_device_certification_reqs_dependent`).
[ ] Pre‑validation run in the CSA Test Harness completed (optional but recommended) with the same PICS and ``dmp-test-skip.xml``.

> [ ] Declaration of Conformity drafted, referencing the Platform Certification ID and listing HW/SW/FW versions.

Reviewer 2: ```suggestion
[ ] Declaration of Conformity (DoC) drafted, referencing the Platform Certification ID and listing hardware, software, and firmware versions.
```

[ ] Submission bundle assembled for the ATL (PICS, ``dmp-test-skip.xml``, DoC, Certification Application ID).
[ ] Submission bundle assembled for the CSA (PICS, ``dmp-test-skip.xml``, DoC, security attestation, dependent certs).
[ ] Provide information to the DCL for the product certification (see :ref:`ug_matter_device_dcl`).

.. note::

>    All necessary document templates (like DoC, Security Attestation, PICS, etc.) for CSA certification submission can be found in the `CSA Matter Resource Kit`_.

Reviewer 2: ```suggestion
   All necessary document templates (like DoC, Security Attestation, PICS) for CSA certification submission can be found in the `CSA Matter Resource Kit`_.
```

Responsibilities and deliverables
*********************************

The following responsibilities and deliverables apply to the platform provider and the product maker.

Platform provider deliverables
=============================

The platform provider must supply the following:

* Platform Certification ID and access for ATL to the platform test report (if necessary).
* SOE specification (supported boards/radios, SDK/OS versions).

> * |NCS| SDK release mapping to the Matter software version.

Reviewer 2: ```suggestion
* |NCS| release mapping to the Matter software version.
```

* Dependent certification references applicable to the platform.
* Security attestation (pre‑filled with platform details).

Product maker deliverables
===============================

As a product maker, you supply the following:

* CSA membership.

> * DoC referencing the Platform Certification ID and listing product HW/SW/FW versions.

Reviewer 2: ```suggestion
* DoC referencing the Platform Certification ID and listing product hardware, software, and firmware versions.
```

* Product PICS with ``PLAT.CERT.TESTS.DONE = True``.
* ``dmp-test-skip.xml`` generated by the PICS Tool.
* Matter Security Attestation (platform + product).
* Dependent certification evidence (Thread/Wi‑Fi/Bluetooth/Ethernet as applicable).

FAQ
***

Can you enable optional clusters not used by the platform?
   Yes, if within platform scope. The ATL includes the relevant cluster test cases in your DMP run.

What if you need a Wi‑Fi or Thread feature the platform did not certify?
   That exceeds platform scope. Request a platform update or recertification first, then proceed with DMP certification.

Do you need to rerun platform tests?
   No. Platform‑only tests are skipped via ``dmp-test-skip.xml``. You still rerun both tests and all product‑specific tests.

Which Matter versions can you use?
   Use the same Matter spec version as the platform.

Where is the SDK?
   `sdk-nrf`_ (use the platform’s certified tag).

Where can you get the platform security attestation template and PICS baseline?

>    Request the templated security attestation via DevZone and download the Platform PICS baseline from the CSA website e.g. ????.

Reviewer 2: ```suggestion
   Request the templated security attestation through `DevZone`_ and download the Platform PICS baseline from the CSA website (add link).
```
